tests/integration/test_charm_scheduled_events.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Integration tests for github-runner charm with scheduled events.
+
+Scheduled events can interfere with most tests. Therefore, an application with
+scheduled events are in its own module.
+"""
+
+
+from asyncio import sleep
+
+import pytest
+from juju.application import Application
+from juju.model import Model
+
+from runner_manager import RunnerManager
+from tests.integration.helpers import (
+    check_runner_binary_exists,
+    get_runner_names,
+    run_in_unit,
+    wait_till_num_of_runners,
+)
+from tests.status_name import ACTIVE_STATUS_NAME
+
+
+@pytest.mark.asyncio
+@pytest.mark.abort_on_fail
+async def test_update_interval(model: Model, app_scheduled_events: Application) -> None:
+    """
+    arrange: An working application with one runner.
+    act:
+        1. Remove runner binary.
+        2. Wait for 3 minutes.
```

Comment:
Out of curiosity, may I ask where 3 minutes comes from?

Reply:
The intervals for scheduled events are 2 mins with a bit of random delay. So 3 mins was a safe choice.